atomic_reactor/plugins/pre_add_help.py

```diff
@@ -66,6 +69,33 @@ def run(self):
             result['status'] = self.NO_HELP_FILE_FOUND
             return result
 
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        labels = dockerfile.labels
+
+        with open(help_path, 'r+') as help_file:
+            try:
```

Comment:
Thought I added a comment about this but I can't find it.

IMHO we shouldn't even use `try:` here. If opening and/or rw-ing the file doesn't work for some reason, I think the build should just fail.

Reply:
Yes you did but with statement handles only file closing but what in case that theres IOError which can occur we should catch it so the whole plugin doesn't fail.

Reply:
I tried it of course and it works maybe you're using python < 2.6 which doesn't support this feature. But as I looked into travis tasks it tests only python 2.6 and greater so olders shouldn't be supported, should they?

Reply:
What I'm suggesting is that we take the `try:` out, as well as the entire `except ...:` block, and just have:
```
with open(...) as ...:
    # read it
    if changes_needed:
        # adjust it
        ...
```

If adjusting that file fails, I think the plugin should fail. Maybe others disagree (@vrutkovs, @lcarva ?)

Reply:
Agree that it should fail the plugin.

Reply:
It seems if the resulting help file cannot be successfully written during build it's probably broken - the build should fail then IMHO

This metadata doesn't look critical for man pages, it just makes the display nicer, right?

Reply:
ok, this is more logical way. it seems I misunderstanded some concepts of plugins. I'll fix that.